Python/fileutils.c

```diff
@@ -2273,6 +2275,7 @@ _Py_normpath(wchar_t *path, Py_ssize_t size)
         *p2++ = *p1++;
         minP2 = p2;
         lastC = SEP;
+        is_absolute = 1;
```

Comment:
Again, if we set `minP2` here to point at the second slash, it'll act like an absolute path without the need for more state. But that isn't how we currently treat it, though I'm not sure that we treat it right at the moment anyway.

Technically, this is probably best treated like two leading slashes on Windows, but only preserving one segment instead of two. At least, that's based on my reading of the spec. I've never seen these used in real life (apart from Blender using them subtly differently), so I don't know what the correct expectation would be.

Reply:
The POSIX spec says the following in [ 4.13 Pathname Resolution](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap04.html#tag_04_13):

> If a pathname begins with two successive <slash> characters, the first component following the leading <slash> characters may be interpreted in an implementation-defined manner, although more than two leading <slash> characters shall be treated as a single <slash> character.

Is this implying that "//.." actually should not be normalized as "//"? In Linux, a "//" root is effectively the same as "/". For example, `os.stat('//..')` queries the root directory.

Reply:
I have no idea what it implies. Linux could be using `//` to imply that the following segment is the top-level directory by that name, but if we're following that rule we should normalise all paths starting with `//` to start with `/` and we don't.

If we're allowing paths to use `//` as defined, we shouldn't be doing anything to change the first segment, because we don't know how it's going to be interpreted.

But that's not the release blocking bug we're facing here, and I'm not aware of anyone raising the issue about it before, so let's just get the smallest fix in to unblock the release and can dive into obscure correctness cases later.

Reply:
Regarding the posix spec: It's better for us to aim for strict compatibility with prior Python versions than new interpretations of specs without a practical bug being filed about something not matching a spec exactly causing a logic problem that prevents a desirable use case and on which platform(s).

Otherwise behavior change creates churn for all users who have code depending on past Python stdlib existing behaviors (whether they know it or not or explicitly test for it or not). Hyrum's Law applies here.

Reply:
I brought it up because it was suggested to shift `minP2` past the first component. If so, `minP2` would have to be special cased for "//.." if it should normalize as "//", per the existing test case. But using "//x" as the root isn't workable anyway because normalizing "//etc/.." as "//etc" wouldn't match the behavior of Linux. 

In Cygwin and MSYS2, "//" is used for UNC paths. In MSYS2, I confirmed that the stat result for "//" is different from the result for "/"; that the stat results for "//", "//..", and "//localhost/.." are all the same; and that the stat result for "//localhost/C$/.." is the same as the result for "//localhost".  This is compatible with the current implementation of `normpath()`.

The following UNIX stack exchange question discusses current and defunct systems and applications that special case "//".

https://unix.stackexchange.com/questions/256497/on-what-systems-is-foo-bar-different-from-foo-bar